### PR TITLE
feat: GSuite detections for admins modifying the trusted domains list and for admins executing a data export

### PR DESCRIPTION
**Pull request by edyesed**
### Background

GSuite Workspace admins can modify the list of trusted domains

### Changes

* 

### Testing

* `make test` 

```shell
user@computer:panther-analysis $ pipenv run panther_analysis_tool test --filter RuleID=GSuite.Workspace.DataExportCreated
[INFO]: Testing analysis items in .

GSuite.Workspace.DataExportCreated
        [PASS] Workspace Admin Data Export Created
                [PASS] [rule] true
                [PASS] [title] GSuite Workspace Data Export [CUSTOMER_TAKEOUT_CREATED] performed by [admin@example.io]
                [PASS] [dedup] GSuite Workspace Data Export [CUSTOMER_TAKEOUT_CREATED] performed by [admin@example.io]
        [PASS] Workspace Admin Data Export Succeeded
                [PASS] [rule] true
                [PASS] [title] GSuite Workspace Data Export [CUSTOMER_TAKEOUT_SUCCEEDED] performed by [admin@example.io]
                [PASS] [dedup] GSuite Workspace Data Export [CUSTOMER_TAKEOUT_SUCCEEDED] performed by [admin@example.io]
        [PASS] Admin Set Default Calendar SHARING_OUTSIDE_DOMAIN Setting to MANAGE_ACCESS
                [PASS] [rule] false

--------------------------
Panther CLI Test Summary
        Path: .
        Passed: 1
        Failed: 0
        Invalid: 0

user@computer:~/Sources/PAN/panther-analysis $ pipenv run panther_analysis_tool test --filter RuleID="GSuite.Workspace.TrustedDomainsAllowlist"
[INFO]: Testing analysis items in .

GSuite.Workspace.TrustedDomainsAllowlist
        [PASS] Workspace Admin Remove Trusted Domain
                [PASS] [rule] true
                [PASS] [title] GSuite Workspace Trusted Domains Modified [REMOVE_TRUSTED_DOMAINS] with [evilexample.com] performed by [user@example.io]
                [PASS] [dedup] GSuite Workspace Trusted Domains Modified [REMOVE_TRUSTED_DOMAINS] with [evilexample.com] performed by [user@example.io]
        [PASS] Workspace Admin Add Trusted Domain
                [PASS] [rule] true
                [PASS] [title] GSuite Workspace Trusted Domains Modified [ADD_TRUSTED_DOMAINS] with [evilexample.com] performed by [user@example.io]
                [PASS] [dedup] GSuite Workspace Trusted Domains Modified [ADD_TRUSTED_DOMAINS] with [evilexample.com] performed by [user@example.io]
        [PASS] Admin Set Default Calendar SHARING_OUTSIDE_DOMAIN Setting to MANAGE_ACCESS
                [PASS] [rule] false

--------------------------
Panther CLI Test Summary
        Path: .
        Passed: 1
        Failed: 0
        Invalid: 0

```

### CODE REVIEW

**Comment by jpl5280** [comment]
Suggestion:

`GSuite Workspace Data Export [CUSTOMER_TAKEOUT_CREATED] performed by [admin@example.io]`

to 

`GSuite Workspace Data Export performed by [admin@example.io]`

as it's redundant/not particularly helpful

File: rules/gsuite_activityevent_rules/gsuite_workspace_trusted_domains_allowlist.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    return event.get("type") == "DOMAIN_SETTINGS" and event.get("name", "").endswith(
+        "_TRUSTED_DOMAINS"
+    )
+
+
+def title(event):
+    return (
+        f"GSuite Workspace Trusted Domains Modified "
+        f"[{event.get('name', '<NO_EVENT_NAME>')}] "
+        f"with [{deep_get(event, 'parameters', 'DOMAIN_NAME', default='<NO_DOMAIN_NAME>')} "
```
Comment (jpl5280) [inline rules/gsuite_activityevent_rules/gsuite_workspace_trusted_domains_allowlist.py:14]: you forgot the closing ]

**Review by jpl5280 — CHANGES_REQUESTED** [review]

**Comment by jpl5280** [comment]
Not seeing the code for it, but you forgot the []'s for `Admin Set Default Calendar SHARING_OUTSIDE_DOMAIN Setting to MANAGE_ACCES`

**Comment by edyesed** [comment]
> Not seeing the code for it, but you forgot the []'s for `Admin Set Default Calendar SHARING_OUTSIDE_DOMAIN Setting to MANAGE_ACCES`

Could this be name of the test vs the [title output](https://github.com/panther-labs/panther-analysis/blob/master/rules/gsuite_activityevent_rules/gsuite_workspace_calendar_external_sharing.py#L21-L24)? When I run tests against that RuleID, I see the text above. It appears the title does have the square bracketing around the interpolated variables. 

```
        [PASS] Admin Set Default Calendar SHARING_OUTSIDE_DOMAIN Setting to MANAGE_ACCESS
                [PASS] [rule] true
                [PASS] [title] GSuite workspace setting for default calendar sharing was changed by [example@example.io] from [READ_WRITE_ACCESS] to [MANAGE_ACCESS]
                [PASS] [dedup] GSuite workspace setting for default calendar sharing was changed by [example@example.io] from [READ_WRITE_ACCESS] to [MANAGE_ACCESS]
```

File: rules/gsuite_activityevent_rules/gsuite_workspace_data_export_created.yml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+RuleID: GSuite.Workspace.DataExportCreated
+DisplayName: GSuite Workspace Data Export Has Been Created
+Enabled: true
+LogTypes:
+  - GSuite.ActivityEvent
+Tags:
+  - GSuite
+Severity: Medium
+Description: >
+  A Workspace Admin Has Created a DataExport
```
Comment (jpl5280) [inline rules/gsuite_activityevent_rules/gsuite_workspace_data_export_created.yml:12]:
```suggestion
  A Workspace Admin Has Created a Data Export
```

File: rules/gsuite_activityevent_rules/gsuite_workspace_data_export_created.yml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+Reference: https://developers.google.com/admin-sdk/reports/v1/appendix/activity/data-studio#DATA_EXPORT
+Runbook: >
+  Verify the intent of this DataExport. If intent cannot be verified, then
```
Comment (jpl5280) [inline rules/gsuite_activityevent_rules/gsuite_workspace_data_export_created.yml:15]:
```suggestion
  Verify the intent of this Data Export. If intent cannot be verified, then
```

File: rules/gsuite_activityevent_rules/gsuite_workspace_data_export_created.yml [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  an indicator search on the actor is advised.
```
Comment (jpl5280) [inline rules/gsuite_activityevent_rules/gsuite_workspace_data_export_created.yml:16]: I'm not sure an indicator search makes sense?

**Review by andrea-youwakim — APPROVED** [review]

**Pull request merged**